README.md

```diff
@@ -0,0 +1,117 @@
+
+We wrote a simple test class using Espresso to check if the text shown to the user is the one obtained from the production code dependency or our test double.
+
+You can find it in ```MainActivityTest.kt```. If you check one of the tests you can discover how we are using Kodein to override the dependencies.
+
+```java
+@Mock private lateinit var applicationScopeClass: ApplicationScopeClass
+
+@Test
+    fun shouldShowInjectedNameProvidedByTheDomainWhenReplaceByAMock() {
+        whenever(applicationScopeClass.getText()).thenReturn("Mock Name")
+        startActivity()
+
+        onView(withText("Mock Name")).check(matches(isDisplayed()))
+    }
+```
+
+We are creating a  Mock (we are using Mockito to provide test doubles, this is not mandatory, you 
+could use your own test doubles, or using other mocking frameworks). We are replacing the real call
+ to getText() for a mock value called 'Mock Name". We launch the activity and we use Expresso to verify that the mocked test has been displayed on the screen.
+
+For this test works perfectly we need to use Kodein to some magic happens. If you continue reading the class at the end of the file you can find the Kodein provisioning dependencies.
+
+```java
+    val testDependencies = Module(allowSilentOverride = true) {
+        bind<ApplicationScopeClass>() with instance(applicationScopeClass)
+        bind<ActivityScopeClass>() with instance(activityScopeClass)
+    }
+```
+
+we create a Module that allows overriding the graph, and we create provisioning for those classes. Depends on your objective you can use instance or provision method. In our case, we used the instance because we are initing Mockito in the test environment setup.
+
+The final part overrides the production graph for the testing ones if you read the setup method you can find how we are doing that.
+
+```java
+   @Before
+    fun setup() {
+        MockitoAnnotations.initMocks(this)
+        val app = InstrumentationRegistry.getInstrumentation().targetContext.asApp()
+        app.resetInjection()
+        app.overrideModule = testDependencies
+    }
+```
+
+First of all, we are initiating all the provisioning mocks with Mockito. After that, we use an extension method that we have in the sample Android application to obtain the application. First, we clean the graph, we like have all dependencies clean before each test, to be repeatable. At last, we save our test module to replace the regular module dependencies.
+
+We need to save the module because we don't have control over the injections. the order of modules has been injected is important. We like that first injected the application modules, after that the activity modules and in the end, the testing modules overriding the previous provisions. When we invoke the start activity, Kodein is going to create the module and injected it in the graph, and we need be ready to apply our testing replaces.
+
```

Comment:
I'd write these two paragraphs as:
```
First of all, we are instantiating all the mocks we are going to use as part of this test using Mockito. Then, we use an extension method that we have in the sample Android application to obtain the application instance. Cleaning  the graph we get a clean environment before each test. This helps us to create repeatable scenarios.

We need to save the module because we don't have control over the injections. Take into account that the order of modules is important. First, we inject the application modules, after that the activity modules and in the end, the testing modules overriding the previous provisions if needed. When we invoke the activity under test, Kodein is going to create the module and inject it into the graph.
```

Reply:
done